### PR TITLE
feat(aws-codepipeline): #2572 allow IAM Role to be passed to Pipeline

### PR DESCRIPTION
**Pull request by AlexChesters**
### Pull Request Checklist

* [N/A] Testing
  - As this feature is very similar to the existing pattern of [artifacts bucket](https://github.com/awslabs/aws-cdk/blob/cfe46f697b1837bf61bec5b950494c26044282ff/packages/%40aws-cdk/aws-codepipeline/lib/pipeline.ts#L224-L234), which isn't unit tested as far as I can see, I didn't add, modify or remove any existing unit tests.
* [X] Docs
  - __jsdocs__: All public APIs documented
  - __README__: README and/or documentation topic updated
  - __Design__: For significant features, design document added to `design` folder
* [X] Title and Description
  - __Change type__: title prefixed with **fix**, **feat** and module name in parens, which will appear in changelog
  - __Title__: use lower-case and doesn't end with a period
  - __Breaking?__: last paragraph: "BREAKING CHANGE: <describe what changed + link for details>"
  - __Issues__: Indicate issues fixed via: "**Fixes #xxx**" or "**Closes #xxx**"
* [N/A] Sensitive Modules (requires 2 PR approvers)
  - IAM Policy Document (in @aws-cdk/aws-iam)
  - EC2 Security Groups and ACLs (in @aws-cdk/aws-ec2)
  - Grant APIs (only if not based on official documentation with a reference)

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

File: packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts [hunk 1]
```diff
@@ -233,9 +239,14 @@ export class Pipeline extends PipelineBase {
...
-      assumedBy: new iam.ServicePrincipal('codepipeline.amazonaws.com')
-    });
+    // If a role has been provided, use it - otherwise, create a role.
+    let propsRole = props.role;
+    if (!propsRole) {
+      propsRole = new iam.Role(this, 'Role', {
+        assumedBy: new iam.ServicePrincipal('codepipeline.amazonaws.com')
+      });
+    }
+    this.role = propsRole;
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts]: We usually write it in one line:

```typescript
this.role = props.role || new iam.Role(this, 'Role', {
  assumedBy: new iam.ServicePrincipal('codepipeline.amazonaws.com')
});
```

File: packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts [hunk 2]
```diff
@@ -68,6 +68,12 @@ export interface PipelineProps {
    */
   readonly artifactBucket?: s3.IBucket;
 
+  /**
+   * The IAM role to be assumed by this Pipeline.
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts]: Can we be a little more precise here? "The IAM role used for executing this Pipeline. If not specified, a new IAM role, trusting the CodePipeline service principal, will be created".

**Comment by AlexChesters** [comment]
Oops, I messed up the history on this branch :| I'll create a new PR..

**Comment by AlexChesters** [comment]
Opened new PR in #2587 (cc: @skinny85)

**Pull request closed without merging**